src/vs/workbench/contrib/terminal/browser/media/shellIntegration-env.zsh

```diff
@@ -2,13 +2,12 @@
 #   Copyright (c) Microsoft Corporation. All rights reserved.
 #   Licensed under the MIT License. See License.txt in the project root for license information.
 # ---------------------------------------------------------------------------------------------
-VSCODE_ZDOTDIR=$ZDOTDIR
-if [[ -f ~/.zshenv ]]; then
-	. ~/.zshenv
-fi
-if [[ "$ZDOTDIR" != "$VSCODE_ZDOTDIR" ]]; then
+if [[ -f $USER_ZDOTDIR/.zshenv ]]; then
+	VSCODE_ZDOTDIR=$ZDOTDIR
+	ZDOTDIR=$USER_ZDOTDIR
+
+	. $USER_ZDOTDIR/.zshenv
+
 	USER_ZDOTDIR=$ZDOTDIR
 	ZDOTDIR=$VSCODE_ZDOTDIR
-else
-	USER_ZDOTDIR=$HOME
 fi
```

Comment:
Am I understanding this right in that it fixes the case where `$ZDOTDIR` (eg. `/home/user/myzsh`) is set in your profile/login scripts and then have a `$ZDOTDIR/.zshenv` in that, not in `~/.zshenv`?

A comment explaining what's going on here would be good.

Reply:
@Tyriar I believe actually what this solves for is when `~/.zshenv` is in a `$ZDOTDIR` which is not equal to `~` as we previously assumed it to be

Reply:
which is why the user's `ZDOTDIR` must be captured before these scripts get run

Reply:
So `ZDOTDIR` gets set in `~/.profile`, or somewhere else I'm not thinking of?

Reply:
set outside of zsh rc files
https://github.com/microsoft/vscode/issues/159785#issuecomment-1241088547

Reply:
> @Tyriar I believe actually what this solves for is when `~/.zshenv` is in a `$ZDOTDIR` which is not equal to `~` as we previously assumed it to be

Exactly